azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -1354,6 +1341,31 @@ public static String getLabelSelector(final Props azkProps) {
     return selectorBuilder.toString();
   }
 
+  /**
+   * Given a flow ID, return the callable endpoint of the flow pod's service where the flow is executed on. The endpoint
+   * is accessible by another pod which resides in the same k8s cluster.
+   * Naming rule of the endpoint: "[serviceName].[namespace]:[servicePort]"
+   * The returned endpoint is presented by a {@link Pair} consisting of the host name and port.
+   *
+   * @param azkProps the Azkaban props
+   * @param executionId the execution ID
+   * @return a {@link Pair} consisting of the endpoint host and port
+   */
+  public static Pair<String, Integer> getFlowServiceEndpoint(final Props azkProps, int executionId) {
```

Comment:
nit, final keywords for executionid. (I think everything that not going to be changed would be final here.

Reply:
thanks for catching it!